### ISSUE TITLE
Gene cluster viewer

### ISSUE THREAD

**Issue by greenify** [biojs/biojs#116]
(migrated from the BioJS wishlist)

"from Genbank files? these fancy arrows"

@bgruening

**Comment by romitgithub** [comment]
Hi Seb, can you explain this in much detail so that i can know what exactly do we need to build and can proceed on this.

**Comment by greenify** [comment]
Hi @romitgithub,

We use often arrows for genes to show whether they are encoded upstream or down stream. Here is an [excellent example](https://www.biostars.org/p/19009/) for this, here is another [one](http://jb.asm.org/content/186/9/2880/F1.expansion.html).

The [UCSC Genome browser doc](http://genome.ucsc.edu/goldenpath/help/hgTracksHelp.html) explains many of the conventions that are used in details.
Here is an excerpt:
> Coding exons are represented by blocks connected by horizontal lines representing introns. The 5' and 3' untranslated regions (UTRs) are displayed as thinner blocks on the leading and trailing ends of the aligning regions. In full display mode, arrowheads on the connecting intron lines indicate the direction of transcription. In situations where no intron is visible (e.g. single-exon genes, extremely zoomed-in displays), the arrowheads are displayed on the exon block itself.
 
Normally those annotations are provided in the [GFF format](http://genome.ucsc.edu/FAQ/FAQformat.html#format3) ([biojs-io-gff](http://biojs.io/d/biojs-io-gff)), but they can also be part of a [Genbank](http://www.ncbi.nlm.nih.gov/Sitemap/samplerecord.html) entry.

I hope this clarifies the idea - don't hesitate to ask again ;-)

**Comment by romitgithub** [comment]
So what i got from this is that the user must be able to upload his sequence and then we mush show him the gene cluster view.

**Comment by greenify** [comment]
>  that the user must be able to upload his sequence 

BioJS component are widgets - you don't need to provide a file upload bar. This is either be part of the embedding website or a component host like the BioJS dashboard. Have a look at these [common code patterns](http://edu.biojs.net/details/code_patterns/)

**Comment by romitgithub** [comment]
Yes that's true, but the essence is that whatever code i right, it must be able to convert a given sequence to a gene cluster view, right?

**Comment by greenify** [comment]
> essence is that whatever code i right, it must be able to convert a given sequence to a gene cluster view, right?

Yes absolutely, but you don't need to reinvent the wheel and write the yet another file upload handler ;-)

**Comment by romitgithub** [comment]
Ok. Thanks for the help :)
